extensions-contrib/oracle-metadata-storage/src/main/java/io/druid/metadata/OracleMetadataConnector.java

```diff
@@ -0,0 +1,306 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.metadata;
+
+import com.google.common.base.Supplier;
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Inject;
+import io.druid.java.util.common.StringUtils;
+import io.druid.java.util.common.logger.Logger;
+import org.apache.commons.dbcp2.BasicDataSource;
+import org.skife.jdbi.v2.DBI;
+import org.skife.jdbi.v2.Handle;
+
+public class OracleMetadataConnector extends SQLMetadataConnector
+{
+  private static final Logger log = new Logger(OracleMetadataConnector.class);
+  private static final String PAYLOAD_TYPE = "BLOB";
+  private static final String SERIAL_TYPE = "NUMBER";
+  private static final String QUOTE_STRING = "\"";
+
+  private final DBI dbi;
+
+  private final Supplier<MetadataStorageConnectorConfig> config;
+  private final Supplier<MetadataStorageTablesConfig> tablesConfigSupplier;
+
+  @Inject
+  public OracleMetadataConnector(
+      Supplier<MetadataStorageConnectorConfig> config,
+      Supplier<MetadataStorageTablesConfig> dbTables
+  )
+  {
+    super(config, dbTables);
+
+    this.config = config;
+    this.tablesConfigSupplier = dbTables;
+
+    final BasicDataSource datasource = getDatasource();
+    // Oracle driver is classloader isolated as part of the extension
+    // so we need to help JDBC find the driver
+    datasource.setDriverClassLoader(getClass().getClassLoader());
+    datasource.setDriverClassName("oracle.jdbc.OracleDriver");
```

Comment:
A lot of the other metadata extensions have a dependency on the driver in the `pom.xml`. I'm not familiar with `oracle.jdbc.OracleDriver`, is there a way to include `ojdbc6.jar` as a pom dependency so that the code at https://github.com/druid-io/druid/pull/5053/files#diff-14475584c3507d73c5b83884ff10d901R57 can pick it up easily without extra setup needed by the administrator?

Reply:
ojdbc jars are not available in maven central. Oracle maintains their own maven repository and developers need to register to access them. Everytime need to provide the user credentials to access it. So I kept them outside of pom.xml because user credentials will be different for different developers. 
Here's their saying about their maven repo: https://docs.oracle.com/middleware/1213/core/MAVEN/config_maven_repo.htm#MAVEN9012 

It's just easier to download the jar and add it in classpath.